adb_graphics/figures/maps.py

```diff
@@ -372,6 +378,34 @@ def _draw_contours(self, ax, not_labeled):
                     print(f'Cannot add contour labels to map for {self.field.short_name} \
                             {self.field.level}')
 
+    def _draw_scatter(self, ax):
+
+        ''' Plot dots at locations on the map that meet a threshold. '''
+
+        field = self.field
+        levels = self.field.clevs
+        alpha_val = 0.75 # set opacity
+        vals = self.field.values()
```

Comment:
It might be good to put the size logic here. Maybe something like:

```
sizes = np.full_like(self.field.values(), np.log10(vals) * 20, dtype='float')
```

Using `full_like` will not cause problems when you change the values. The data is not linked back to reference those saved in `self.field`.

[This example](https://www.pythonprogramming.in/adjust-marker-sizes-and-colors-in-scatter-plot.html) shows that you might be able to provide the size and color as `s` and `c` args in the call to `scatter`.

Reply:
Matplotlib's scatter plot method is limited.  To do the scatter plot on a map background requires the field, which has the lat,lon (x,y) info.  The scatter plot method will use the values associated with the field as the size field, even if the sizes are provided to the method in a 1-dimensional format (also required by the scatter plot method) the same size as the colors array.  If I leave "field=field" and add the sizes field, it will crash on having two instances of the size array.  If I try to set "field=sizes", it will crash because it doesn't have the lat,lon information.  This is why the field needs to be manipulated after the colors have been set using the original values, and then the values can be manipulated to adjust the sizes.